src/NumericalAlgorithms/Spectral/SwshInterpolation.cpp

```diff
@@ -101,9 +101,9 @@ void SpinWeightedSphericalHarmonic::evaluate(
                      pow(sin_theta_over_2,
                          2 * static_cast<int>(l_) - (2 * r + spin_ - m_));
     } else if (2 * static_cast<int>(l_) < 2 * r + spin_ - m_) {
-      theta_factor = pow(cos_theta_over_2 / sin_theta_over_2,
-                         2 * r + spin_ - m_ - 2 * static_cast<int>(l_)) *
-                     pow(cos_theta_over_2, 2 * l_);
+      ASSERT(equal_within_roundoff(gsl::at(r_prefactors_, r), 0.),
```

Comment:
If you want to do these checks, use an exact comparison.  Assigning a literal 0.0 does not incur roundoff error.

I think it would be cleaner to change the upper bound of the loop to `r <= static_cast<int>(l_) + min(m_, -spin_)` so that this branch is never even entered.  Then the remaining if branch has the normal expression in it and the else branch can be commented as existing just to avoid a `pow(0, 0)`.